src/abstract_models/device_model.jl

```diff
@@ -0,0 +1,26 @@
+abstract type AbstractDeviceFormulation end
+
+mutable struct DeviceModel{D <: PSY.PowerSystemDevice,
+                           B <: PSI.AbstractDeviceFormulation}
+    device::Type{D}
+    formulation::Type{B}
+end
+
+function construct_device!(ps_m::CanonicalModel,
+                           device_model::DeviceModel,
+                           system_formulation::Type{S},
+                           system::PSY.PowerSystem,
+                           time_range::UnitRange{Int64};
+                           kwargs...) where {S <: PM.AbstractPowerFormulation}
+
+    construct_device!(ps_m,
+                      device_model.device,
+                      device_model.formulation,
+                      system_formulation,
+                      system,
+                      time_range;
+                      kwargs...)
+    
+    return nothing
```

Comment:
The style guide (redirection to JuMP) says to use "return" instead of "return nothing".
http://www.juliaopt.org/JuMP.jl/dev/style/#Return-statements-1

Reply:
Fixed via 4917588